Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -51,12 +64,19 @@ public class NotificationPublisher : INotificationPublisher {
       languageServer.TextDocument.SendNotification(DafnyRequestNames.VerificationSymbolStatus, notification);
     }
 
-    private static FileVerificationStatus? GetFileVerificationStatus(DafnyDocument document) {
-      if (document.ImplementationIdToView == null || document.VerificationTasks == null) {
+    private static FileVerificationStatus? GetFileVerificationStatus(CompilationView document) {
+      if (!document.ImplementationsWereUpdated) {
+        /*
+         CompilationAfterResolution.Snapshot() gets migrated ImplementationViews.
+         It has to get migrated Diagnostics inside ImplementationViews, otherwise we get incorrect diagnostics.
+         However, migrating the ImplementationId's may mean we lose verifiable symbols, which we don't want at this point. TODO: why not?
```

Comment:
TODO here. I'm not sure I can help since I don't understand this paragraph. Could you clarify a bit?

Reply:
I find the need for the boolean `ImplementationsWereUpdated` surprising. I think we may be able to get rid of it but I didn't want to spend time on that in this PR. I added what I know about it in this comment.

Notifications are only sent if they're different than the previous notification, but currently a migrated notification, that is semantically the same as the previous one, is also considered different. This boolean prevents verification symbol notification from being sent before translation has been done, since they are never useful to the client.